core/node/commitment_generator/src/lib.rs

```diff
@@ -263,22 +266,45 @@ impl CommitmentGenerator {
             }
             state_diffs.sort_unstable_by_key(|rec| (rec.address, rec.key));
 
-            let blob_commitments = if protocol_version.is_post_1_4_2() {
+            let blob_hashes = if protocol_version.is_post_1_4_2() {
                 let pubdata_input = header.pubdata_input.with_context(|| {
                     format!("`pubdata_input` is missing for L1 batch #{l1_batch_number}")
                 })?;
 
-                pubdata_to_blob_commitments(num_blobs_required(&protocol_version), &pubdata_input)
+                let commitments = pubdata_to_blob_commitments(
+                    num_blobs_required(&protocol_version),
+                    &pubdata_input,
+                );
+                let linear_hashes = pubdata_to_blob_linear_hashes(
+                    num_blobs_required(&protocol_version),
+                    pubdata_input,
+                );
+
+                commitments
+                    .into_iter()
+                    .zip(linear_hashes)
+                    .map(|(commitment, linear_hash)| BlobHash {
+                        commitment,
+                        linear_hash,
+                    })
+                    .collect::<Vec<_>>()
             } else {
-                vec![H256::zero(); num_blobs_required(&protocol_version)]
+                vec![Default::default(); num_blobs_required(&protocol_version)]
             };
 
+            let mut connection = self
+                .connection_pool
+                .connection_tagged("commitment_generator")
+                .await?;
+            let aggregation_root = read_aggregation_root(&mut connection, l1_batch_number).await?;
```

Comment:
Dumb question: Won't this root be always zero and unused for `protocol_version < Gateway`? Would not reading it from storage in this case complicate the code too much?

Reply:
yes, it will be zero. No, it won't complicate, makes sense to wrap it into if